tests/Aql/ExecutionBlockImplTest.cpp

```diff
@@ -391,6 +401,925 @@ TEST_F(ExecutionBlockImplTest,
+    EXPECT_EQ(output.numRowsLeft(), 20);
+    while (inputRange.hasDataRow() && !output.isFull()) {
+      auto const& [state, input] = inputRange.nextDataRow();
+      EXPECT_TRUE(input.isInitialized());
+      // We do not write anything, so just advance!
+      output.copyRow(input);
+      output.advanceRow();
+    }
+
+    NoStats stats{};
+    // Forward the client call.
+    return {inputRange.upstreamState(), stats, output.getClientCall()};
+  };
+
+  ExecutionBlockImpl<LambdaExePassThrough> testee{fakedQuery->engine(), node,
+                                                  makeInfos(std::move(execImpl))};
+  auto singleton = createSingleton();
+  testee.addDependency(singleton.get());
+  auto [state, skipped, block] = testee.execute(stack);
+
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(skipped, 1);
+  EXPECT_EQ(block, nullptr);
+  // Once with empty, once with the line by Singleton
+  EXPECT_EQ(nrCalls, 0);
+}
+
+TEST_F(ExecutionBlockImplExecuteTest, test_toplevel_offset_only_call_execute) {
+  AqlCall fullCall{};
+  fullCall.offset = 20;
+  fullCall.softLimit = 0;
+  AqlCallStack stack{std::move(fullCall)};
+  ExecutionNode const* node = nullptr;
+  size_t nrCalls = 0;
+
+  ProduceCall execImpl = [&nrCalls](AqlItemBlockInputRange& inputRange, OutputAqlItemRow& output)
+      -> std::tuple<ExecutorState, LambdaExePassThrough::Stats, AqlCall> {
+    auto const& clientCall = output.getClientCall();
+    if (nrCalls > 10) {
+      EXPECT_TRUE(false);
+      // This is emergency bailout, we ask way to often here
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL);
+    }
+    nrCalls++;
+    EXPECT_EQ(clientCall.getOffset(), 0);
+    EXPECT_FALSE(std::holds_alternative<AqlCall::Infinity>(clientCall.softLimit));
+    EXPECT_EQ(std::get<std::size_t>(clientCall.softLimit), 20);
+    EXPECT_TRUE(std::holds_alternative<AqlCall::Infinity>(clientCall.hardLimit));
+    EXPECT_EQ(output.numRowsLeft(), 20);
+
+    while (inputRange.hasDataRow() && !output.isFull()) {
+      auto const& [state, input] = inputRange.nextDataRow();
+      EXPECT_TRUE(input.isInitialized());
+      // We do not write anything, so just advance!
+      output.copyRow(input);
+      output.advanceRow();
+    }
+
+    NoStats stats{};
+    return {inputRange.upstreamState(), stats, output.getClientCall()};
+  };
+
+  ExecutionBlockImpl<LambdaExePassThrough> testee{fakedQuery->engine(), node,
+                                                  makeInfos(std::move(execImpl))};
+  auto singleton = createSingleton();
+  testee.addDependency(singleton.get());
+  auto [state, skipped, block] = testee.execute(stack);
+
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(skipped, 1);
+  EXPECT_EQ(block, nullptr);
+  // Once with empty, once with the line by Singleton
+  EXPECT_EQ(nrCalls, 0);
+}
+
+enum CallAsserterState { INITIAL, SKIP, GET, COUNT, DONE };
+
+// Asserts if AqlCalls are forwarded onmodified correctly.
```

Comment:
is this supposed to be "unmodified"? Also this sounds as if an assertion is raised if the the calls are forwarded correctly.